### PR TITLE
Data download script (and associated instructions)

### LINKED ISSUE #169: Make data download script for development purposes
Make a script that downloads the data from the S3 bucket for holding example data.
Probably will store this in the `scripts` folder. Will need to add instructions of how to use it in CONTRIBUTING.md guidelines.

### PR DESCRIPTION
**Pull request by cansavvy**
## Purpose:
Closes #169 

## Strategy
There's a `scripts/download-data.sh` executable script. It uses aws client so I added that to the Dockerfile. It was easier to do that so I could copy over the whole folders and its contents. 

I have the Snakefile run the download-data.sh script but I am unsure if I did it correctly?

I've also added instructions to CONTRIBUTING.md

## Concerns/Questions for reviewers:
- Does this work? 
- Are there instructions clear? There are no screenshots. Can add if needed.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends apt-utils dialo
 
 # Add curl, bzip2 and some dev libs
 RUN apt-get update -qq && apt-get -y --no-install-recommends install \
-    curl \
```
Comment (jashapiro) [inline Dockerfile:11]: If curl is not there to start, why take it out? Could be useful!

Comment (cansavvy) [reply Dockerfile:11]: Okay, I was figuring if we didn't use it for downloading the data, then we probably don't need it. But I'll add it back.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -130,7 +193,7 @@ For example: `GSE12345_pca_plot.png`
 - Output results should be `TSV` when possible.  
 - Plots should be saved to `PNG` whenever possible.    
 - Underscores instead of `-` just so we don't have to think about it.
-- No camel case. 
+- No camel case.
```
[the remark this answers is not in the document]
Comment (cansavvy) [reply CONTRIBUTING.md:196]: Good point.

**Comment by jaclyn-taroni** [comment]
I had the same general concerns about running the S3 download as much or as little as required based on whether or not the files changed.

**Comment by cansavvy** [comment]
After trying the ideas @jashapiro noted above: https://github.com/AlexsLemonade/refinebio-examples/pull/180#pullrequestreview-468420271 , we realized it's probably just easier/better to download the data ourselves outside of snakemake rather than having snakemake run it for us but not necessarily make smart decisions about when to run it. The CONTRIBUTING.md instructions still apply to this situation.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -112,13 +152,36 @@ After a few rounds of changes to the Dockerfile has occurred, we can make a new
...
+6) Choose the `metadata` and `data` TSV files you obtained from refine.bio (make sure they are correctly non-QN'ed or QN'ed upon your download).
+7) Click `Next`.
+8) Under `Manage public permissions` > Choose `Grant public read access to this object`.
+9) Click `Next`.
+10) Click `Next` again.
+11) Click Upload.
+
+Test that the files you've uploaded succesfully download by running the `download-data.sh`
+script.
+If you run the script and it says `Access Denied` you may have missed step 8, but you can go back to the  file and click the `Make it Public` button and try testing your download again.
```
Comment (cbethell) [inline CONTRIBUTING.md:175]:
```suggestion
If you run the script and it says `Access Denied` you may have missed step 8, but you can go back to the file and click the `Make it Public` button and try testing your download again.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -155,7 +155,7 @@ So the same steps would be followed above, but you can change where it says `dev
 ### Adding datasets to the S3 bucket
 
 You will need an AWS account to add files to S3 bucket.
-Go to the [refinebio-examples bucket](https://s3.console.aws.amazon.com/s3/buckets/refinebio-examples/02-microarray/?region=us-east-2&tab=overview) on Amazon Web Services.
+Go to the [refinebio-examples bucket](https://s3.console.aws.amazon.com/s3/buckets/refinebio-examples/?region=us-east-2) on Amazon Web Services.
```
Comment (jashapiro) [inline CONTRIBUTING.md:158]:
```suggestion
Go to the [refinebio-examples bucket](https://s3.console.aws.amazon.com/s3/buckets/refinebio-examples/) on Amazon Web Services.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -24,6 +56,13 @@ docker run --mount type=bind,target=/home/rstudio,source=$PWD -e PASSWORD=<PASSW
 Now you can navigate to http://localhost:8787/ to start developing.
 Login to the RStudio server with the username `rstudio` and the password you set above.
 
+## Download the datasets
+
+For developing purposes, you can download all the datasets for examples by running this command in the Docker container:
```
Comment (jashapiro) [inline CONTRIBUTING.md:61]:
```suggestion
For development purposes, you can download all the datasets for the example notebooks by running this command in the Docker container:
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -112,13 +152,36 @@ After a few rounds of changes to the Dockerfile has occurred, we can make a new
 For example, if there is a `ccdl/refinebio-examples:v1` we can move to `v2`.
 So the same steps would be followed above, but you can change where it says `dev` with the appropriate `v` number.
 
+### Adding datasets to the S3 bucket
+
+You will need an AWS account to add files to S3 bucket.
+Go to the [refinebio-examples bucket](https://s3.console.aws.amazon.com/s3/buckets/refinebio-examples/) on Amazon Web Services.
+
+Each section has a folder (`02-microarray`, `03-rnaseq`, and `04-advanced-topics`).
+1) Click on the section folder you are adding to.
```
Comment (jashapiro) [inline CONTRIBUTING.md:161]:
```suggestion
1) Click on the folder for the section you are adding to, then on the `data` subfolder.
```

**Pull request merged**